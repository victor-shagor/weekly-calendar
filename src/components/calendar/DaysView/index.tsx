import { format, isEqual, isToday } from "date-fns";
import classNames from "classnames";
import { daysOfTheWeek, DaysViewProps } from "../../../constants";

function DaysView({ daysArray, selectedDay, handleClick }: DaysViewProps) {
  return (
    <div data-testid="daysview-container">
      <div
        style={{
          display: "flex",
          margin: "1rem",
          justifyContent: "space-between",
        }}
      >
        {daysOfTheWeek.map((day, idx) => (
          <div>{day}</div>
        ))}
      </div>
      <div
        data-testid="day-grid"
        style={{
          display: "flex",
          margin: "0.3rem",
          justifyContent: "space-between",
        }}
      >
        {daysArray.map((day: any) => (
          <div key={day.toString()}>
            <button
              type="button"
              onClick={() => handleClick(day)}
              className={classNames(
                isEqual(day, selectedDay) && "white",
                !isEqual(day, selectedDay) && isToday(day) && "blue",
                isEqual(day, selectedDay) && !isToday(day) && "bg-dark",
                isEqual(day, selectedDay) && isToday(day) && "bg-blue",
                "btn-date"
              )}
            >
              <time dateTime={format(day, "yyyy-MM-dd")}>
                {format(day, "d")}
              </time>
            </button>
          </div>
        ))}
      </div>
    </div>
  );
}

export default DaysView;
