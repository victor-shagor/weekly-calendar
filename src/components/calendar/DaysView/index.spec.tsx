import { render, fireEvent, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import DaysView from ".";

const mockDaysArray = [
  new Date(),
  new Date("2022-10-01T23:00:00.000Z"),
  new Date("2022-10-02T23:00:00.000Z"),
];

const handleClick = jest.fn();
describe("days view", () => {
  it("should render days view", () => {
    render(
      <DaysView
        handleClick={handleClick}
        selectedDay={mockDaysArray[2]}
        daysArray={mockDaysArray}
      />
    );

    const btnDiv = screen.getByTestId("day-grid");

    expect(handleClick).toHaveBeenCalledTimes(0);
    expect(btnDiv.querySelector(".btn-date")).toBeInTheDocument();
    expect(btnDiv.querySelector(".bg-dark")).toBeInTheDocument();
  });

  it("should call handleClick function", () => {
    render(
      <DaysView
        handleClick={handleClick}
        selectedDay={mockDaysArray[0]}
        daysArray={mockDaysArray}
      />
    );
    const btnClick = screen
      .getByTestId("day-grid")
      .querySelectorAll(".btn-date")[0];
    fireEvent.click(btnClick);
    expect(handleClick).toHaveBeenCalledTimes(1);
  });
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <DaysView
          handleClick={handleClick}
          selectedDay={mockDaysArray[0]}
          daysArray={mockDaysArray}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
