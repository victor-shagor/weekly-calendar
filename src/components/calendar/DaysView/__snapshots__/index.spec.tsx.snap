// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`days view renders correctly 1`] = `
<div
  data-testid="daysview-container"
>
  <div
    style={
      Object {
        "display": "flex",
        "justifyContent": "space-between",
        "margin": "1rem",
      }
    }
  >
    <div>
      S
    </div>
    <div>
      M
    </div>
    <div>
      T
    </div>
    <div>
      W
    </div>
    <div>
      T
    </div>
    <div>
      F
    </div>
    <div>
      S
    </div>
  </div>
  <div
    data-testid="day-grid"
    style={
      Object {
        "display": "flex",
        "justifyContent": "space-between",
        "margin": "0.3rem",
      }
    }
  >
    <div>
      <button
        className="white bg-blue btn-date"
        onClick={[Function]}
        type="button"
      >
        <time
          dateTime="2022-10-27"
        >
          27
        </time>
      </button>
    </div>
    <div>
      <button
        className="btn-date"
        onClick={[Function]}
        type="button"
      >
        <time
          dateTime="2022-10-02"
        >
          2
        </time>
      </button>
    </div>
    <div>
      <button
        className="btn-date"
        onClick={[Function]}
        type="button"
      >
        <time
          dateTime="2022-10-03"
        >
          3
        </time>
      </button>
    </div>
  </div>
</div>
`;
